boards/xtensa/m5stack_core2/m5stack_core2.dts

```diff
@@ -0,0 +1,254 @@
+/*
+ * Copyright (c) 2023 Martin Kiepfer <mrmarteng@teleschirm.org>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+/dts-v1/;
+
+#include <espressif/esp32/esp32_common.dtsi>
+#include "m5stack_core2-pinctrl.dtsi"
+#include "grove_connectors.dtsi"
+#include <zephyr/dt-bindings/display/ili9xxx.h>
+#include <zephyr/dt-bindings/regulator/axp192.h>
+
+/ {
+	model = "esp32";
+	compatible = "espressif,esp32";
+
+	aliases {
+		pwr-led = &pwr_led;
+		uart-0 = &uart0;
+		i2c-0 = &i2c0;
+		watchdog0 = &wdt0;
+		rtc = &pfc8563_rtc;
+	};
+
+	chosen {
+		zephyr,sram = &sram0;
+		zephyr,console = &uart0;
+		zephyr,shell-uart = &uart0;
+		zephyr,flash = &flash0;
+		zephyr,display = &ili9342c;
+		zephyr,keyboard-scan = &kscan_input;
+		zephyr,code-partition = &slot0_partition;
+		zephyr,rtc = &pfc8563_rtc;
+	};
+};
+
+&cpu0 {
+	clock-frequency = <ESP32_CLK_CPU_240M>;
+};
+
+&cpu1 {
+	clock-frequency = <ESP32_CLK_CPU_240M>;
+};
+
+&psram0 {
+	reg = <0x3f800000 DT_SIZE_M(8)>;
+	status = "disabled";
+};
+
+&uart0 {
+	status = "okay";
+	current-speed = <115200>;
+	pinctrl-0 = <&uart0_tx_gpio1 &uart0_rx_gpio3>;
+	pinctrl-names = "default";
+};
+
+&uart1 {
+	status = "disabled";
+	current-speed = <115200>;
+	pinctrl-0 = <&uart1_rx_gpio33 &uart1_tx_gpio32>;
+	pinctrl-names = "default";
+};
+
+&gpio0 {
+	status = "okay";
+};
+
+&gpio1 {
+	status = "okay";
+};
+
+&i2c0 {
+	status = "okay";
+	clock-frequency = <I2C_BITRATE_STANDARD>;
+	sda-gpios = <&gpio0 21 GPIO_OPEN_DRAIN>;
+	scl-gpios = <&gpio0 22 GPIO_OPEN_DRAIN>;
+	pinctrl-0 = <&i2c0_default>;
+	scl-timeout-us = <0>;
+	pinctrl-names = "default";
+
+	pfc8563_rtc: pfc8563@51 {
+		compatible = "nxp,pcf8523";
+		reg = <0x51>;
+		status = "okay";
+		battery-switch-over = "disabled";
+	};
+
+	axp192_pmic: axp192@34 {
+		compatible = "x-powers,axp192";
+		reg = <0x34>;
+		status = "okay";
+
+		axp192_regulator: axp192_regulator {
+			compatible = "x-powers,axp192-regulator";
+			status = "okay";
+
+			vdd_mcu: DCDC1 {
+				regulator-init-microvolt = <3350000>;
+				regulator-min-microvolt = <3200000>;
+				regulator-max-microvolt = <3400000>;
+				regulator-initial-mode = <AXP192_DCDC_MODE_AUTO>;
+				regulator-boot-on;
+				regulator-always-on;
+			};
+
+			lcd_bg: DCDC3 {
+				regulator-init-microvolt = <2800000>;
+				regulator-min-microvolt = <0>;
+				regulator-max-microvolt = <2800000>;
```

Comment:
i would align with the min/max values from the Arduino core: min = 2500mV, max = 3300mV

https://github.com/m5stack/M5Core2/blob/master/src/AXP192.cpp#L450-L453

In any case I should say it's pretty cool to be able to tinker with the regulator shell commands to set the brightness of the LCD or disable it altogether.

Reply:
fixed with laters commit